test/intentctl/test_intentfiles.py

```diff
@@ -0,0 +1,227 @@
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+
+from base import IntentCtlTestBase
+import json
+import os
+
+from intentctl import intentfiles
+
+
+class IntentStoreTests(IntentCtlTestBase):
+
+    def test_new_intent_store(self):
+        """
+        A smoke test to ensure nothing bizarre happens on IntentStore object creation
+        :return:
+        """
+        temp_dir = os.path.join(self._mktmp(), 'intentfile.json')
+        intent_store = self.assertRaisesNothing(intentfiles.IntentStore, temp_dir)
+        self.assertEqual(intent_store.contents, {})
+        self.assertEqual(intent_store.path, temp_dir)
+
+    def test_read_file_non_existent_file(self):
+        """
+        Can the IntentStore.read_file method handle attempting to read a file which does not exist?
+        :return:
+        """
+        temp_dir = os.path.join(self._mktmp(), 'intentfile.json')
+        self.assertFalse(os.path.exists(temp_dir))
+
+        intent_store = self.assertRaisesNothing(intentfiles.IntentStore, temp_dir)
+        res = self.assertRaisesNothing(intent_store.read_file)
+        self.assertFalse(bool(res))
+
+    def _read_file(self, file_contents=None, expected_contents=None):
+        """
+        Utility method for logic common to the *read_file* tests.
+        :param file_contents:
+        :param expected_contents:
+        :return:
+        """
+        temp_dir = os.path.join(self._mktmp(), 'intentfile.json')
+        with open(temp_dir, 'w') as f:
+            if file_contents and not isinstance(file_contents, str):
+                json.dump(file_contents, f)
+            else:
+                f.write(file_contents or '')
+            f.flush()
+        self.assertTrue(os.path.exists(temp_dir), "Unable to create test file in temp dir")
+
+        # Actually do the test
+        intent_store = self.assertRaisesNothing(intentfiles.IntentStore, temp_dir)
+        self.assertRaisesNothing(intent_store.read_file)
+        self.assertEqual(intent_store.contents, expected_contents)
+
+    def test_read_file_empty_file(self):
+        """
+        Can the IntentStore.read_file method handle attempting to read an empty file?
+        :return:
+        """
+        self._read_file(file_contents='', expected_contents={})
+
+    def test_read_file_non_empty(self):
+        """
+        Lets see if we can read a file that is non-empty
+        :return:
+        """
+        test_data = {"arbitrary": "data"}
+        self._read_file(file_contents=test_data, expected_contents=test_data)
+
+    def test_create(self):
+        """
+        Verify that the create method will create the directory (if needed), and that the resulting
+        file in the directory is writable by us.
+        :return:
+        """
```

Comment:
This test should be modified. You should provide path. The method should be able to create directory and then you should test that the file is file and you can write to this file.